app/javascript/dashboard/helper/specs/messageHelper.spec.js

```diff
@@ -0,0 +1,136 @@
+import {
+  replaceVariablesInMessage,
+  getFirstName,
+  getLastName,
+  getMessageVariables,
+  getUndefinedVariablesInMessage,
+} from '../messageHelper';
+
+const variables = {
+  'contact.name': 'John Doe',
+  'contact.first_name': 'John',
+  'contact.last_name': 'Doe',
+  'contact.email': 'john.p@example.com',
+  'contact.phone': '1234567890',
+  'conversation.id': 1,
+  'agent.first_name': 'Samuel',
+  'agent.last_name': 'Smith',
+  'agent.email': 'samuel@gmail.com',
+};
+
+describe('#replaceVariablesInMessage', () => {
+  it('returns the message with variable name', () => {
+    const message =
+      'No issues. Hey {{contact.first_name}}, we will send the reset instructions to your email {{ contact.email}}. The {{ agent.first_name }} {{ agent.last_name }} will take care of everything. Your conversation id is {{ conversation.id }}.';
+    expect(replaceVariablesInMessage({ message, variables })).toBe(
+      'No issues. Hey John, we will send the reset instructions to your email john.p@example.com. The Samuel Smith will take care of everything. Your conversation id is 1.'
+    );
+  });
```

Comment:
Are we allowing customers to make templates? Assuming we do, should we also validate if the number of opening and closing brackets always match? 

Say the customer adds the following as a template, should we be flagging these?

```
Hey {{contact.first_name} this is chatwoot support, 
{{agent.first_name}} will get back to you soon
```

Reply:
No. We don't support variables in incoming messages (https://github.com/chatwoot/chatwoot/pull/6077/files#diff-4dd1799f11dc16334e627bfdcc3fc10ed11cf0a6c7b3fdb230641cf2704cace8R21)

Reply:
I meant, even the users of Chatwoot dashboard, when they create these templates, should we validate the brackets?

Reply:
It will be nice to have.

Reply:
Added https://github.com/chatwoot/chatwoot/pull/6236 to prevent any breakages in the backend.